### ISSUE TITLE
Disable tgz extraction on pull

### ISSUE THREAD

**Issue by daniel-pebble** [oras-project/oras-py#119]
It seems that client.pull() automatically extracts archives which is a nice feature however I don't require it so it is possible to disable?

I've also noticed that the result from the pull is an invalid path, the path is actually the path to the extacted folder however the returned path includes the tgz which is infact in another folder (all be it with a random name)

![image](https://github.com/oras-project/oras-py/assets/47243318/84fc6625-08da-4f5c-a3b0-00a0246c8bab)

![image](https://github.com/oras-project/oras-py/assets/47243318/9b1a2e00-b21b-4cc1-b0af-316b8fffb599)

![image](https://github.com/oras-project/oras-py/assets/47243318/5c874eb4-6de5-4b54-bf3b-3a75dc45fa31)

**Comment by vsoch** [comment]
It's partially related to this issue: https://github.com/oras-project/oras-go/issues/644. We should possibly follow what they decide to do and then mimic it here.

**Comment by daniel-pebble** [comment]
When I use "oras pull" it doesn't unpack the tgz automatically so I thought it was just an issue with the py library.
Looking at the pull function, could we add an argument like "skip_unpack= kwargs.get("skip_unpack", false)" and then check the value of that variable here
[            if layer["mediaType"] == oras.defaults.default_blob_dir_media_type:](https://github.com/oras-project/oras-py/blob/88bad1ffa7d339871a24ec6c37a7b1020f06cf8f/oras/provider.py#L817)

**Comment by vsoch** [comment]
Yes absolutely - I think that's the direction the upstream client will go to. Would you care to do a pull request? We can add the argument, and ensure that:

1. Documentation for usage is added
2. A test
3. Update the CHANGELOG.md and version.py

Thank you!
